### PR TITLE
Implement tests for `(when)` option

### PR DESCRIPTION
**Pull request by yevhenii-nadtochii**
During the migration to typed expressions, we discovered an incompatible expression in `InTimeGenerator`. It was fixed in-place. This PR adds missing tests for `(when)` option and fixes another bug, which led to generation of validation code for Spine `Temporal` when the message declared `repeated google.protobuf.Timestamp`.

Now, there are two similar test suites to cover Protobuf `Timestamp` and Spine `Temporal` (particularly `io.spine.time.LocalDateTime`). Repeated fields have also been covered.

### CODE REVIEW

**Comment by yevhenii-nadtochii** [comment]
@armiol @alexander-yevsyukov PTAL

File: java-tests/validating/src/test/kotlin/io/spine/test/options/when/ProtoTimestampWhenSpec.kt [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+ * Protobuf [Duration] of fifty milliseconds.
+ *
+ * To shift the time into the past or future, we add or subtract a difference of this amount.
+ *
+ * There are two reasons for choosing fifty milliseconds:
+ *
+ * 1. The generated code uses `io.spine.base.Time.currentTime()` to get the current timestamp
+ * for comparison. In turn, this method relies on `io.spine.base.Time.SystemTimeProvider`
+ * by default, which has millisecond precision.
+ * 2. Adding too small amount of time to make the stamp denote "future" might be unreliable.
```
Comment (armiol) [inline java-tests/validating/src/test/kotlin/io/spine/test/options/when/ProtoTimestampWhenSpec.kt:239]: So why don't we choose two days instead?

File: java-tests/validating/src/test/kotlin/io/spine/test/options/when/SpineTemporalWhenSpec.kt [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+ * Fifty milliseconds.
+ *
+ * To shift the time into the past or future, we add or subtract a difference of this amount.
+ *
+ * There are two reasons for choosing fifty milliseconds:
+ *
+ * 1. The generated code uses `io.spine.base.Time.currentTime()` to get the current timestamp
+ * for comparison. In turn, this method relies on `io.spine.base.Time.SystemTimeProvider`
+ * by default, which has millisecond precision.
+ * 2. Adding too small amount of time to make the stamp denote "future" might be unreliable.
```
Comment (armiol) [inline java-tests/validating/src/test/kotlin/io/spine/test/options/when/SpineTemporalWhenSpec.kt:240]: Same question as the one above.

**Review by armiol — COMMENTED** [review]
In general, LGTM. However, please see my question(s).

File: java-tests/validating/src/test/kotlin/io/spine/test/options/when/ProtoTimestampWhenSpec.kt (hunk 1 above)
Comment (yevhenii-nadtochii) [reply java-tests/validating/src/test/kotlin/io/spine/test/options/when/ProtoTimestampWhenSpec.kt:239]: To be closer to the precision threshold. `two days` is already a seconds-based precision. Any value less than 1s would confirm millisecond precision during comparison.

**Review by alexander-yevsyukov — CHANGES_REQUESTED** [review]
Please see my comments.

File: java-tests/validating/src/test/kotlin/io/spine/test/options/when/ProtoTimestampWhenSpec.kt (hunk 1 above)
Comment (yevhenii-nadtochii) [reply java-tests/validating/src/test/kotlin/io/spine/test/options/when/ProtoTimestampWhenSpec.kt:237]: I've bumped to 500ms in case of slow VMs on CI.

**Review by armiol — APPROVED** [review]

**Review by alexander-yevsyukov — APPROVED** [review]
LGTM

**Pull request merged**